server/service/schedule/schedule.go

```diff
@@ -199,6 +203,27 @@ func (s *Schedule) Start() {
 						ctxerr.Handle(s.ctx, err)
 					}
 				}
+
+				totalRunningTime := time.Since(newStart)
+				schedInterval = s.getSchedInterval()
+
+				if totalRunningTime > schedInterval {
+					level.Info(s.logger).Log("cron", s.name, "msg", "total runtime (%d) exceeded schedule interval (%d)")
+				}
+
+				// set new wait to the remaining duration after dividing the total running time by the schedule interval
+				newWait = schedInterval - (totalRunningTime % schedInterval)
+				setWaitTimes(newStart, newWait)
+
+				select {
+				case <-schedTicker.C:
```

Comment:
I suppose my question is because I'm not understanding why we do this here.

Reply:
It is possible that tick was sent on the channel while the run was in progress in the case that the runtime is longer than the interval. So here we're doing a non-blocking select to grab that tick (if any) before we exit this iteration of the loop. If there's nothing on the tick channel, we'll just move on to cancel the hold and start the next iteration.

Reply:
But in that case, wouldn't it be more likely that some other instance grabbed it? You've already went over.

Reply:
No because the lock gets refreshed  the entire time it was running (thanks to the holdLock)

Reply:
The other instances will stay in sync because of the, but this particular instance needs to have the ticker re-synced

Reply:
Ah, right. Thanks for helping me understand.